[DOCS] Building and deploying EMT on edge devices (#363)
open-edge-platform/edge-microvisor-toolkit: docs/developer-guide/get-started/emt-build-and-deploy.md

**Table of contents:**

1. [Build a custom image of Edge Microvisor Toolkit.](#1-build-the-microvisor)
2. [Create a bootable USB installer.](#2-create-a-bootable-usb-installer)
3. [Install the toolkit on an edge device.](#3-install-the-toolkit-on-an-edge-device)
4. [Deploy AI solution.](#4-deploy-ai-solution)

## 1. Build the microvisor

If you want to create your own custom standalone node, follow the instructions below to build
the microvisor from scratch. If you wish to use the default, non-real-time image of Edge Microvisor
skip to [Create a bootable USB installer](#2-create-a-bootable-usb-installer).

1. Clone the stable branch of edge-microvisor-toolkit repository:

   ```bash
   git clone https://github.com/open-edge-platform/edge-microvisor-toolkit --branch=3.0.20250718
   ```

2. Navigate to the `edge-microvisor-toolkit` directory:

   ```bash
   cd edge-microvisor-toolkit
   ```

3. Install prerequisites:

   > **Note:** Requirements for building the toolchain were validated on Ubuntu 22.04 host.

   ```bash
   # Install required dependencies.
   sudo ./toolkit/docs/building/prerequisites-ubuntu.sh

>    # Also supported is:

Reviewer: ```suggestion
   # Also supported:
```

   ```

   Download the Kubernetes artifacts (container images and manifest files) by
   executing the `./download_images.sh` script. For the default non-real-time image,
   use the `NON-RT` parameter.

   ```bash
   sudo ./download_images.sh NON-RT
   ```

   > **Note:** By default, the script will only pull basic kubernetes artifacts to create a
     single node cluster. If you want to use EMT image with desktop virtualization features,
     use the `DV` parameter instead of `NON-RT`.

   Update the `config-file` with your deployment-specific settings.
   This configuration file is used to provision the edge node during
   its initial boot and should include the following parameters:

   - **Proxy settings** - when the edge node requires a proxy to access external networks.
   - **SSH key** - the public SSH key (typically your `id_rsa.pub`) from your Linux
     development system to enable passwordless SSH access to the edge node.
   - **User credentials** - the username and password for the primary user account on the edge node.
   - **Cloud-init customization** - optional, user-defined `cloud-init` configurations for advanced setup requirements.

>      For the default non-real-time image, a basic Kubernetes installation is done automatically.

Reviewer: ```suggestion
     For the default non-RT image, a basic Kubernetes installation is done automatically.
```

# Build Edge Microvisor Toolkit and Deploy on Edge Nodes

> This guide will walk you through the entire process of deploying Edge Microvisor Toolkit.

Reviewer 2: ```suggestion
This guide will walk you through the entire process of deploying the Edge Microvisor Toolkit.
```

You will learn how to build the toolkit, install it on single or multiple edge node
clusters, and deploy AI solutions.

The Edge Microvisor Toolkit Standalone Node supports different EMT images to
meet specific edge deployment needs. You can choose between:

- **Edge Microvisor Toolkit Non-Real-Time image** (default)
- **Edge Microvisor Toolkit Desktop Virtualization image**
- **Custom immutable Edge Microvisor Toolkit created using** [Build Instructions](#1-build-the-microvisor)

**Table of contents:**

1. [Build a custom image of Edge Microvisor Toolkit.](#1-build-the-microvisor)
2. [Create a bootable USB installer.](#2-create-a-bootable-usb-installer)
3. [Install the toolkit on an edge device.](#3-install-the-toolkit-on-an-edge-device)
4. [Deploy AI solution.](#4-deploy-ai-solution)

## 1. Build the microvisor

If you want to create your own custom standalone node, follow the instructions below to build

> the microvisor from scratch. If you wish to use the default, non-real-time image of Edge Microvisor

Reviewer 2: ```suggestion
the Edge Microvisor from scratch. If you wish to use the default, non-real-time image of Edge Microvisor
```

skip to [Create a bootable USB installer](#2-create-a-bootable-usb-installer).

> 1. Clone the stable branch of edge-microvisor-toolkit repository:

Reviewer 2: Perhaps, state as an example using a tagged version such as, as this will change on a week by week basis. Perhaps tell the user how to list all the tagged releases and pick the latest one?

   For more information about specific building parameters, refer to [this article](https://github.com/open-edge-platform/edge-microvisor-toolkit/blob/3.0/toolkit/docs/building/building.md#local-build-variables).

## 2. Create a bootable USB installer

### Prepare the bootable USB drive

Ensure the correct USB drive is selected to avoid data loss.
Replace `/dev/sdX` with the device name of your USB drive.

- Insert the USB drive and identify the USB disk:

   ```bash
   lsblk -o NAME,MAJ:MIN,RM,SIZE,RO,FSTYPE,MOUNTPOINT,MODEL
   ```

- Use the `wipefs` command to remove any existing filesystem signatures from the USB drive.
  This ensures a clean slate for formatting

   ```bash
   sudo wipefs --all --force /dev/sdX
   ```

- Format the USB drive with the FAT32 filesystem, using `mkfs.vfat`:

   ```bash
   sudo mkfs.vfat /dev/sdX
   ```

- Unmount the USB drive to ensure the creation of bootable USB.

  - Check what is currently mounted:

    ```bash
    df -hT
    ```

  - Unmount the drive:

    ```bash
    sudo umount /dev/sdX
    ```

### Prepare the installation files

1. Clone the edge-microvisor-toolkit-standalone-node repository:

   ```bash
   cd

>    git clone https://github.com/open-edge-platform/edge-microvisor-toolkit-standalone-node

Reviewer 2: You should have similar cloning of a tagged release as opposed to the tip of the tree for consistence.

   Since the `./standalone-node/installation-scripts/out` folder is read only, you need to
   select a folder to which you will extract the required files and then edit them.
   For example, you can create the `standalone-installation-files` folder in the
   `\home` directory:

   ```bash
   mkdir ~\standalone-installation-files
   ```

   Extract the contents to the selected folder:

   ```bash
   tar -xzf ./standalone-node/installation_scripts/out/standalone-installation-files.tar.gz -C ~/standalone-installation-files
   ```

   The extracted files are as follows:

   ```text
   bootable-usb-prepare.sh
   write-image-to-usb.sh
   config-file
   usb-bootable-files.tar.gz
   edgenode-logs-collection.sh
   standalone-vm-setup.sh
   download_images.sh
   user-apps
   ```

   Navigate to the folder with the extracted installation files:

   ```bash
   cd ~/standalone-installation-files
   ```

   Download the Kubernetes artifacts (container images and manifest files) by
   executing the `./download_images.sh` script. For the default non-real-time image,
   use the `NON-RT` parameter.

   ```bash
   sudo ./download_images.sh NON-RT
   ```

>    > **Note:** By default, the script will only pull basic kubernetes artifacts to create a

Reviewer 2: ```suggestion
   > **Note:** By default, the script will only pull basic Kubernetes artifacts to create a
```

     single node cluster. If you want to use EMT image with desktop virtualization features,
     use the `DV` parameter instead of `NON-RT`.

   Update the `config-file` with your deployment-specific settings.
   This configuration file is used to provision the edge node during
   its initial boot and should include the following parameters:

   - **Proxy settings** - when the edge node requires a proxy to access external networks.
   - **SSH key** - the public SSH key (typically your `id_rsa.pub`) from your Linux
     development system to enable passwordless SSH access to the edge node.
   - **User credentials** - the username and password for the primary user account on the edge node.

>    - **Cloud-init customization** - optional, user-defined `cloud-init` configurations for advanced setup requirements.

Reviewer 2: ```suggestion
   - **cloud-init customization** - optional, user-defined `cloud-init` configurations for advanced setup requirements.
```

     For the default non-RT image, a basic Kubernetes installation is done automatically.

     > **Note:** For deployments requiring desktop virtualization features,
     > refer to the [desktop virtualization image guide](https://github.com/open-edge-platform/edge-microvisor-toolkit-standalone-node/blob/main/standalone-node/docs/user-guide/desktop-virtualization-image-guide.md) in the
     > `user-guide` directory. This document provides reference `cloud-init` configurations that can be
     > tailored to your specific deployment needs.

>    - **HugePages configuration** - set when the workloads require HugePages.

Reviewer 2: No link or refernce how to enable huge pages support here... is that intended. It seems like something is missing.

Author: It reflects how it's described in the [user guide](https://github.com/sgolebiewski-intel/edge-microvisor-toolkit-standalone-node/blob/main/standalone-node/docs/user-guide/get-started-guide.md#13-prepare-the-bootable-usb-drive) in the standalone-node repository;
